test/parallel_api/algorithm/alg.nonmodifying/transform_if.pass.cpp

```diff
@@ -0,0 +1,165 @@
+        using out_value_type = typename ::std::iterator_traits<OutputIterator>::value_type;
+        // call transform_if
+        oneapi::dpl::transform_if(exec, first, last, mask, result_begin,
+                                  mutable_negate_first<in_value_type1, in_value_type2>{},
+                                  mutable_check_mask_second<in_value_type1, in_value_type2>{});
+
+        //calculate expected
+        std::vector<out_value_type> expected(n);
+        auto in_iter = first;
+        auto mask_iter = mask;
+        auto expected_iter = expected.begin();
+        for (; in_iter != last; in_iter++, mask_iter++, expected_iter++)
+        {
+            *expected_iter = *mask_iter == 1 ? -(*in_iter) : 0;
+        }
+
+        EXPECT_EQ_N(expected.begin(), result_begin, n, "wrong effect from transform_if binary");
+
+        // reset output elements to 0
+        ::std::fill(result_begin, result_end, 0);
+    }
+};
+
+struct test_transform_if_unary
+{
+    template <typename Policy, typename InputIterator1, typename OutputIterator, typename Size>
+    void
+    operator()(Policy&& exec, InputIterator1 first, InputIterator1 last, OutputIterator result_begin,
+               OutputIterator result_end, Size n)
+    {
+        using in_value_type = typename ::std::iterator_traits<InputIterator1>::value_type;
+        using out_value_type = typename ::std::iterator_traits<OutputIterator>::value_type;
+
+        // call transform_if
+        oneapi::dpl::transform_if(exec, first, last, result_begin, mutable_negate<in_value_type>{},
+                                  mutable_check_mod3_is_0<in_value_type>{});
+
+        //calculate expected
+        std::vector<out_value_type> expected(n);
+        auto expected_iter = expected.begin();
+        auto in_iter = first;
+        for (; in_iter != last; in_iter++, expected_iter++)
+        {
+            *expected_iter = *in_iter % 3 == 0 ? -(*in_iter) : 0;
+        }
+
+        EXPECT_EQ_N(expected.begin(), result_begin, n, "wrong effect from transform_if unary");
+
+        // reset output elements to 0
+        ::std::fill(result_begin, result_end, 0);
+    }
+};
+
+template <typename In1, typename In2, typename Out>
+void
+test()
+{
+    for (size_t n = 1; n <= 100000; n = n <= 16 ? n + 1 : size_t(3.1415 * n))
+    {
+        {
+            Sequence<In1> in1(n, [=](size_t k) { return (3 * k); });
+            Sequence<In2> in2(n, [=](size_t k) { return k % 2 == 0 ? 1 : 0; });
+
+            Sequence<Out> out(n, [=](size_t) { return 0; });
+
+            invoke_on_all_policies<0>()(test_transform_if_binary(), in1.begin(), in1.end(), in2.begin(), in2.end(),
+                                        out.begin(), out.end(), n);
+#if !ONEDPL_FPGA_DEVICE
+            invoke_on_all_policies<1>()(test_transform_if_binary(), in1.cbegin(), in1.cend(), in2.cbegin(), in2.cend(),
+                                        out.begin(), out.end(), n);
+#endif
+        }
+        {
```

Comment:
Should we add test cases for an in-place `transform_if`?

Reply:
I've added an inplace test.